core/src/main/java/io/github/tcdl/cli/CliTool.java

```diff
@@ -1,26 +1,122 @@
-import io.github.tcdl.events.Event;
-import io.github.tcdl.events.SingleArgumentAdapter;
-import io.github.tcdl.messages.Message;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.json.JSONObject;
+import io.github.tcdl.adapters.Adapter;
+import io.github.tcdl.adapters.AdapterFactory;
+import io.github.tcdl.config.MsbConfigurations;
+
+import java.io.IOException;
+import java.util.*;
 
 public class CliTool {
-    // TODO get rid of this hardcode
-    private static final String TOPIC_NAME = "search:parsers:facets:v1";
+    private static final AdapterFactory adapterFactory = AdapterFactory.getInstance();
+    private static final MsbConfigurations configuration = MsbConfigurations.msbConfiguration();
+    private static final MsbConfigurations.BrokerAdapter brokerType = configuration.getBrokerType();
+    private static final Set<String> registeredTopics = new HashSet<>();
 
     public static void main(String[] args) {
-        ChannelManager channelManager = ChannelManager.getInstance();
+        // Parse command-line arguments
+        List<String> topics = getOptionAsList(args, "--topic", "-t");
+        if (topics == null || topics.isEmpty()) {
+            System.out.println("Please specify topic via --topic <topic name>");
+            return;
+        }
+
+        List<String> follow = getOptionAsList(args, "--follow", "-f");
+        if (follow == null || follow.isEmpty()) {
+            follow = Collections.singletonList("response");
+        }
+
+        boolean pretty = getOptionAsBoolean(args, "--pretty", "-p");
+
+        // Subscribe to the configured topics
+        for (String topicName : topics) {
+            subscribe(topicName, new CliMessageHandler(pretty, follow));
+        }
+    }
+
+    private static boolean getOptionAsBoolean(String[] args, String optionName, String alias) {
+        for (int i = 0; i < args.length; i++) {
+            String arg = args[i];
+
+            if (arg.equals(optionName) || arg.equals(alias)) {
+                if (i + 1 < args.length) {
+                    String optionValue = args[i + 1];
+
+                    return optionValue.equals("true");
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private static List<String> getOptionAsList(String[] args, String optionName, String alias) {
+        for (int i = 0; i < args.length; i++) {
+            String arg = args[i];
+
+            if (arg.equals(optionName) || arg.equals(alias)) {
+                if (i + 1 < args.length) {
+                    String optionValue = args[i + 1];
+
+                    return Arrays.asList(optionValue.split(","));
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private static void subscribe(String topicName, CliMessageHandler handler) {
+        synchronized (registeredTopics) {
```

Comment:
This code is invoked when we process response messages. Hence it can be called from different threads in parallel. To ensure consistency of the set we have to synchronize the block.

Any other suggestions?